IHP/ControllerSupport.hs

```diff
@@ -87,25 +89,61 @@ runAction controller = do
 
     doRunAction `catches` [ Handler handleResponseException, Handler (\exception -> ErrorController.displayException exception controller "")]
 
-{-# INLINE runActionWithNewContext #-}
-runActionWithNewContext :: forall application controller. (Controller controller, ?applicationContext :: ApplicationContext, ?context :: RequestContext, InitControllerContext application, ?application :: application, Typeable application, Typeable controller) => controller -> IO ResponseReceived
-runActionWithNewContext controller = do
+{-# INLINE runActionWithContext #-}
+runActionWithContext :: forall application controller. (Controller controller, ?modelContext :: ModelContext, ?applicationContext :: ApplicationContext, ?requestContext :: RequestContext) => ControllerContext -> controller -> IO ResponseReceived
+runActionWithContext controllerContext controller = let ?context = controllerContext in runAction controller
```

Comment:
This function IMO is the same as `runAction`. Can't we then just remove this function entirely?

Reply:
Yes, you are absolutely correct. Probably a holdover from some changes I made. Fixed!